test/jit/test_dtype_analysis.py

```diff
@@ -245,3 +320,87 @@ def func(input, weight, bias, y):
         y_val = torch.rand((1,), dtype=torch.float32)
         input_args = [conv_in.input, *conv_in.args, y_val]
         self.assert_dtype_equal_custom_args(func, input_args)
+
+
+class TestDtypeCustomRules(TestDtypeBase):
+    def assert_output_dtype_equal(self, expected_res, prop_graph):
+        actual_dtype = self.node_output_dtype(prop_graph)
+        if len(actual_dtype) == 1:
+            self.assert_tensor_dtype_equal(expected_res, actual_dtype[0])
+        else:
+            self.assertEqual(len(expected_res), len(actual_dtype))
+            for expected, actual in zip(expected_res, actual_dtype):
+                self.assert_tensor_dtype_equal(expected, actual)
+
+    def assert_tensor_dtype_equal(self, tensor_output, graph_dtype):
+        if not isinstance(tensor_output, torch.Tensor):
+            return
+        self.assertEqual(tensor_output.dtype, graph_dtype)
+
+    @ops([op for op in op_db if op.aten_name in custom_rules_works_list])
+    def test_custom_rules(self, device, dtype, op):
+        # Return fn, inputs_fn for all
+        sample_input = op.sample_inputs(device, dtype, requires_grad=False)[0]
+        input_args = [sample_input.input, *sample_input.args]
+        expected_res = op(*input_args, **sample_input.kwargs)
+        func = op.get_op()
+        traced_fn = create_traced_fn(self, func)
```

Comment:
out of curiosity, why not scripting?

Reply:
@eellison I remember that you recommended Tracing here. Could you elaborate here?

Reply:
Tracing more reliably works than scripting at getting IR